post-processor/googlecompute-import/post-processor.go

```diff
@@ -0,0 +1,204 @@
+package googlecomputeimport
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"google.golang.org/api/compute/v1"
+	"google.golang.org/api/storage/v1"
+
+	"github.com/hashicorp/packer/builder/googlecompute"
+	"github.com/hashicorp/packer/common"
+	"github.com/hashicorp/packer/helper/config"
+	"github.com/hashicorp/packer/helper/multistep"
+	"github.com/hashicorp/packer/packer"
+	"github.com/hashicorp/packer/post-processor/compress"
+	"github.com/hashicorp/packer/template/interpolate"
+
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/jwt"
+)
+
+type Config struct {
+	common.PackerConfig `mapstructure:",squash"`
+
+	Bucket            string            `mapstructure:"bucket"`
+	ImageDescription  string            `mapstructure:"image_description"`
+	ImageFamily       string            `mapstructure:"image_family"`
+	ImageLabels       map[string]string `mapstructure:"image_labels"`
+	ImageName         string            `mapstructure:"image_name"`
+	ProjectId         string            `mapstructure:"project_id"`
+	AccountFile       string            `mapstructure:"account_file"`
+	KeepOriginalImage bool              `mapstructure:"keep_input_artifact"`
+
+	ctx interpolate.Context
+}
+
+type PostProcessor struct {
+	config Config
+	runner multistep.Runner
+}
+
+func (p *PostProcessor) Configure(raws ...interface{}) error {
+	err := config.Decode(&p.config, &config.DecodeOpts{
+		Interpolate:        true,
+		InterpolateContext: &p.config.ctx,
+	}, raws...)
+	if err != nil {
+		return err
+	}
+
+	errs := new(packer.MultiError)
+
+	templates := map[string]*string{
+		"bucket":       &p.config.Bucket,
+		"image_name":   &p.config.ImageName,
+		"project_id":   &p.config.ProjectId,
+		"account_file": &p.config.AccountFile,
+	}
+	for key, ptr := range templates {
+		if *ptr == "" {
+			errs = packer.MultiErrorAppend(
+				errs, fmt.Errorf("%s must be set", key))
+		}
+	}
+
+	if len(errs.Errors) > 0 {
+		return errs
+	}
+
+	return nil
+}
+
+func (p *PostProcessor) PostProcess(ui packer.Ui, artifact packer.Artifact) (packer.Artifact, bool, error) {
+	ui.Say("Starting googlecompute-import...")
+
+	if artifact.BuilderId() != compress.BuilderId {
+		err := fmt.Errorf(
+			"incompatible artifact type: %s\nCan only import from Compress post-processor artifacts",
+			artifact.BuilderId())
+		return nil, p.config.KeepOriginalImage, err
```

Comment:
For simplicity "keep artefact" can be `false` when failing.

Reply:
Fixed